src/string/formatter.cr

```diff
@@ -248,8 +248,15 @@ struct String::Formatter(A)
   end
 
   def char(flags, arg) : Nil
+    val = if arg.is_a?(Char)
+      arg
+    elsif arg.is_a?(Int::Primitive)
+      arg.chr
+    else
+      raise ArgumentError.new("Expected a char or integer, not #{arg.inspect}")
+    end
```

Comment:
**polish:** The extra assignment is technically unnecessary. What do you think about rewriting it this way which I find a bit cleaner?

```suggestion
    if arg.is_a?(Int::Primitive)
      arg = arg.chr
    end
    
    unless arg.is_a?(Char)  
      raise ArgumentError.new("Expected a char or integer, not #{arg.inspect}")
    end
```